src/redis-cli.c

```diff
@@ -1576,10 +1576,10 @@ static int parseOptions(int argc, char **argv) {
         } else if (!strcmp(argv[i],"--scan")) {
             config.scan_mode = 1;
         } else if (!strcmp(argv[i],"--pattern") && !lastarg) {
-            if (config.pattern) sds_free(config.pattern);
+            if (config.pattern) sdsfree(config.pattern);
```

Comment:
No need to check `if (config.pattern)`, since [`sdsfree`](https://github.com/redis/redis/blob/ea1b48bd12d4cbd6634e5a478da99c6a1f14be8e/src/sds.c#L193-L197) already takes care of that.
Same as line 1582..

Reply:
I don't like this pattern that assumes freeing a NULL is OK, but it is more consistent with the rest of the code indeed.